l3_dump.py

```diff
@@ -430,6 +424,59 @@ def do_main(args:list, return_logentry_lists:bool = False) -> int:
     print(f"Unpacked {nentries=} log-entries.")
     return (nentries, tid_list, loc_list, msg_list, arg1_list, arg2_list)
 
+# #############################################################################
+def l3_parse_args(args:list):
+    """
+    Parse command-line arguments. Return parsed-arguments object
+    """
+    # ---------------------------------------------------------------
+    # Start of argument parser, with inline examples text
+    # Create 'parser' as object of type ArgumentParser
+    # ---------------------------------------------------------------
+    parser  = argparse.ArgumentParser(description='Unpack L3 log-entries from log-file'
+                                                 + ' generated by named program',
+                                      formatter_class=argparse.RawDescriptionHelpFormatter,
+                                      epilog=r'''Examples:
+
+- Basic usage:
+    ''' + sys.argv[0]
+        + ''' --log-file <L3-log-file> --binary <program-binary>
+
+NOTE: If <program-binary>, built with L3_LOC_ENABLED=1, invokes L3-logging,
+      we expect to find a corresponding LOC-decoder binary named
+      <program-binary>_loc, needed for decoding LOC-ID entries in the log-file.
+''')
+
+    # Define arguments supported by this script
+    parser.add_argument('--log-file', dest='log_file'
+                        , metavar='<log-file-name>'
+                        , required=True
+                        , help='L3 log-file name')
+
+    parser.add_argument('--binary', dest='prog_binary'
+                        , metavar='<program-binary>'
+                        , required=True
+                        , help='Program binary generating L3 logging')
+
+    # ======================================================================
+    # Debugging support
+    parser.add_argument('--verbose', dest='verbose'
+                        , action='store_true'
+                        , default=False
+                        , help='Show verbose progress messages')
+
+    parser.add_argument('--debug', dest='debug_script'
```

Comment:
I added args `--verbose` and `--debug` in anticipation of using them in future. I will add test cases to exercise these in future revisions.